src/sample.ts

```diff
@@ -0,0 +1,80 @@
+import { purry } from './purry';
+
+export interface SampleType {
+  size: number;
+  repeating: boolean;
+}
+
+function _sample<T>(items: Array<T>, options: number | SampleType) {
+  let nbrSamples = typeof options === 'number' ? options : options.size;
+  const repeating = typeof options === 'object' && options?.repeating;
+
+  let size = items.length;
+  items = [...items];
+
+  if (nbrSamples > size && !repeating) nbrSamples = size;
```

Comment:
The second yellow flag is that we will go through sampling the whole array for P > N although the result is effectively a shuffle (in your impl) or a trivial identity (in my impl).

Reply:
Yes, that is correct. And as far as I know the Fisher-Yates' shuffle is still considered to be way to go for that: https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm

Reply:
That's actually not my concern, your impl is fisher-yates enough imho, the concern was that we now have 2 methods that effectively do the same thing (for a specific input). So shuffle === sample(HUGE_SAMPLE_SIZE), and I don't like having this duplication in case someone comes up with an optimization or whatever and doesn't port it to the other case.